clientapi/routing/register_test.go

```diff
@@ -208,3 +212,123 @@ func TestValidationOfApplicationServices(t *testing.T) {
 		t.Errorf("user_id should not have been valid: @_something_else:localhost")
 	}
 }
+
+func TestValidatePassword(t *testing.T) {
+	t.Parallel()
+	defaults := &config.PasswordRequirements{}
+	defaults.Defaults()
+
+	custom := &config.PasswordRequirements{
+		MinPasswordLength: 16,
+		MaxPasswordLength: 32,
+		RequireMixedCase:  true,
+		MinNumberSymbols:  5,
+	}
+	var testCases = []struct {
+		name     string
+		config   config.PasswordRequirements
+		password string
+		expected *util.JSONResponse
+	}{
+		{
+			"default reject too short",
```

Comment:
Minor nitpick: the tests are "hard" to read, without the fields specified.

Reply:
Second this — can you please name the fields in the values?

Reply:
Does something like this work?

```
                        "default reject too short", // test name
                        *defaults,                  // password requirements
                        "foobar",                   // password value to test
                        &util.JSONResponse{ // expected response
                                Code: http.StatusBadRequest,
                                JSON: jsonerror.WeakPassword(fmt.Sprintf("password too weak: min %d chars", defaults.MinPasswordLength)),
                        },
```

Reply:
oh! now i understand. okay, will do.